pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -16,62 +16,93 @@ package ready
 
 import (
 	"fmt"
+	"net"
 
 	admin "github.com/envoyproxy/go-control-plane/envoy/admin/v2alpha"
+	"github.com/hashicorp/go-multierror"
 
 	"istio.io/istio/pilot/cmd/pilot-agent/status/util"
 	"istio.io/istio/pilot/pkg/model"
+	"istio.io/pkg/log"
 )
 
 // Probe for readiness.
 type Probe struct {
-	LocalHostAddr       string
-	NodeType            model.NodeType
-	AdminPort           uint16
-	receivedFirstUpdate bool
+	LocalHostAddr  string
+	NodeType       model.NodeType
+	AdminPort      uint16
+	lastKnownState *probeState
+}
+
+type probeState struct {
+	serverState  uint64
+	versionStats util.Stats
 }
 
 // Check executes the probe and returns an error if the probe fails.
 func (p *Probe) Check() error {
+	// Initialize the ServerState to 2 (PRE_INITIALIZING) because 0 indicates LIVE.
+	if p.lastKnownState == nil {
```

Comment:
Nit: feel free to omit
Does it make sense to put the lastKnowState when initializing `Probe` instead of the `Check`?